packages/orchestration/src/exos/chain-account-kit.js

```diff
@@ -133,39 +133,37 @@ export const prepareChainAccountKit = (zone, { watch, when }) =>
          *
          * @param {AnyJson[]} msgs
          * @param {Omit<TxBody, 'messages'>} [opts]
-         * @returns {Promise<string>} - base64 encoded bytes string. Can be
-         *   decoded using the corresponding `Msg*Response` object.
+         * @returns {Vow<string>} - base64 encoded bytes string. Can be decoded
+         *   using the corresponding `Msg*Response` object.
          * @throws {Error} if packet fails to send or an error is returned
          */
-        async executeEncodedTx(msgs, opts) {
+        executeEncodedTx(msgs, opts) {
           const { connection } = this.state;
           // TODO #9281 do not throw synchronously when returning a promise; return a rejected Vow
           /// see https://github.com/Agoric/agoric-sdk/pull/9454#discussion_r1626898694
           if (!connection) throw Fail`connection not available`;
-          return when(
-            watch(
-              E(connection).send(makeTxPacket(msgs, opts)),
-              this.facets.parseTxPacketWatcher,
-            ),
+          return watch(
```

Comment:
What's the best way to handle synchronous throws in a way that satisfies membrane requirements? Since we can't return promises, should we wrap in a watch?
```js
if(!connection) return watch(Promise.reject(Fail`connection not available`));
```

Reply:
Updated to use `asVow` from #9577